### ISSUE TITLE
[observability] add sync percentage during boot-up

### ISSUE THREAD

**Issue by pavitthrap** [stacks-network/stacks-blockchain#3557]
As @wileyj mentioned in the weekly blockchain sync, it would be helpful to add a sync percentage to provide some visibility of the progress of the block download (Bitcoin has something like this).

**Comment by diwakergupta** [comment]
For reference, the log message in bitcoind looks like this (note the `progress` key)
```
2023-02-06T12:08:21Z UpdateTip: new best=000000000000000000054a90eca11b8c89ca12e9a0f4dee54543d2340a9615d0 height=775266 version=0x2d46a000 log2_work=93.988069 tx=802885497 date='2023-02-06T09:02:12Z' progress=0.999959 cache=69.5MiB(505916txo)
```

**Comment by wileyj** [comment]
the question is where to expose this data though - adding a new log entry might be hard to discern from the other log output. 
using an endpoint could work, but that would likely require state to be maintained somewhere.
if the work is combined with the other issue #1229  it might be easier to expose in a human readable way since the state should already exist for that other issue.

i think adding  a % to the  endpoint in #2768 would be more than sufficient (and should be relatively easy to add to that PR since the data would already exist. just a little extra math to be added).

**Comment by jcnelson** [comment]
Temporarily assigning to @pavitthrap, just so it has an owner.  Please feel free to re-assign or close as you see fit!

**Comment by pavitthrap** [comment]
Closed in #3754
